bft-lib/src/simulated_context.rs

```diff
@@ -71,26 +72,21 @@ impl SimulatedLedgerState {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct SimulatedContext<Config> {
+pub struct SimulatedContext {
     author: Author,
-    config: Config,
+    database: HashMap<String, Vec<u8>>,
     num_nodes: usize,
     max_command_per_epoch: usize,
     next_fetched_command_index: usize,
     last_committed_ledger_state: SimulatedLedgerState,
     pending_ledger_states: HashMap<State, SimulatedLedgerState>,
 }
 
-impl<Config> SimulatedContext<Config> {
-    pub fn new(
-        author: Author,
-        config: Config,
-        num_nodes: usize,
-        max_command_per_epoch: usize,
-    ) -> Self {
+impl SimulatedContext {
+    pub fn new(author: Author, num_nodes: usize, max_command_per_epoch: usize) -> Self {
         SimulatedContext {
             author,
-            config,
+            database: HashMap::new(),
```

Comment:
Do you clean this up at some point? If not, could we run out of memory if we run very long simulations?

Reply:
Good point. The current simulator is memory-only for speed, but we could add a CLI option later to use a real DB.